### ISSUE TITLE
A crawler takes a curated dev docs endpoint

### ISSUE THREAD

**Issue by wsxiaoys** [TabbyML/tabby#791]
Reference: https://github.com/getcursor/docs/blob/main/docs.jsonl

May not necessarily be something need to reimplement, lots of tools in this space:

- https://github.com/projectdiscovery/katana
- https://github.com/rivermont/spidy

**Comment by wsxiaoys** [comment]
Seems following katana snippet does the trick

```
katana -u https://codemirror.net/docs/ -cs docs -fs fqdn -d 1000 -j > output.json

-u: craw starting url
-cs: url path prefix scoping
-fs fqdn: restrict the same domain.
-d: depth
-j: json line output
```

**Comment by wsxiaoys** [comment]
related: https://www.fleet.so/context collect all docs for python

**Comment by akash4566** [comment]
[
  {
    "id": "859e8dff-f9ec-497d-aa07-344e48b2f67b",
    "score": 0.848275101,
    "metadata": {
      "library_id": "4506492b-70de-49f1-ba2e-d65bd7048a28",
      "page_id": "732e264c-c077-4978-bc93-380d7dc28983",
      "parent": "3be9bbcc-b5d6-4a91-9f72-a570c2db33e5",
      "text": "Quickstart | 🦜️🔗 Langchain",
      "url": "https://python.langchain.com/docs/get_started/quickstart"
    }
  },

**Comment by wsxiaoys** [comment]
this will be part of 0.13 release

**Comment by wsxiaoys** [comment]
cc @boxbeam

**Comment by wsxiaoys** [comment]
Released as `web` context provider in 0.13
